File: 8vim/src/main/java/inc/flide/vim8/keyboardHelpers/KeyboardAction.java
package inc.flide.vim8.keyboardHelpers;

import inc.flide.vim8.structures.KeyboardActionType;

public class KeyboardAction {

    private KeyboardActionType keyboardActionType;
    private String text;
    private String capsLockText;
    private int keyEventCode;
    private int keyFlags;

    public KeyboardAction(KeyboardActionType keyboardActionType, String text, String capsLockText, int keyEventCode ,int keyFlags) {
        this.keyboardActionType = keyboardActionType;
        this.text = text;
        this.keyEventCode = keyEventCode;
        setCapsLockText(capsLockText);
        this.keyFlags = keyFlags;
    }

    private void setCapsLockText(String capsLockText) {
        if((capsLockText==null || capsLockText.length()==0) && this.text!=null){
            this.capsLockText = this.text.toUpperCase();
        } else{
            this.capsLockText = capsLockText;
        }
    }

    public int getKeyFlags() { return keyFlags; }

    public String getText() {
        return text;
    }

    public int getKeyEventCode() {
        return keyEventCode;
    }

    public KeyboardActionType getKeyboardActionType() {
        return keyboardActionType;
    }

    public String getCapsLockText() { return capsLockText; }
}

File: 8vim/src/main/java/inc/flide/vim8/structures/KeyboardAction.kt
package inc.flide.vim8.structures

import java.util.Locale

class KeyboardAction(
	val keyboardActionType: KeyboardActionType,
	val text: String?,
	capsLockText: String?,
	val keyEventCode: Int,
	val keyFlags: Int
) {
    val capsLockText: String? = if (capsLockText.isNullOrEmpty() && text != null) {
        text.uppercase(Locale.getDefault())
    } else {
        capsLockText
    }
}
